src/vs/workbench/contrib/positronPlots/browser/components/dynamicPlotInstance.tsx

```diff
@@ -119,6 +119,11 @@ export const DynamicPlotInstance = (props: DynamicPlotInstanceProps) => {
 
 			// If we're rendering, show a progress bar.
 			if (state === PlotClientState.Rendering) {
+				// Before starting a new render, remove any existing progress bars. This prevents
+				// a buildup of progress bars when rendering multiple times and ensures the progress bar
+				// is removed when a new render is requested before the previous one completes.
+				progressRef.current.replaceChildren();
```

Comment:
Should we maintain a single progress bar and ensure that `done()`/`hide()` are called at the correct times rather than deleting and recreating the DOM nodes? This seems fine as is though.

Reply:
I think this might be more complicated with the timing to create the progress bar once the ref to the parent div is available. I'll see if I can follow up with something if it isn't too complicated.